dangvuthang/rocodingtest
# Feat/internal cheating detection

### Changelog

- Added `InstructionBox` component to explain 
- Added custom hooks to handle detection of whether **document is focused**, and web page is displayed in **fullscreen**
- Added `Warning` component to display cheating message
- Added logic to handle cheating detection

### How to test

- Checkout this branch
- Navigate to `/exam` page
- Follow the instruction
- Tried minimize the web page, or change tab or open another window on top of the web page
- See if a message displayed correctly
- Tried cheating 3 times and test if the application kick user back to the main page

close #37
